tests/Doctrine/Tests/DBAL/SQLParserUtilsTest.php

```diff
@@ -356,6 +356,15 @@ public function dataExpandListParameters()
                 array(1, 2),
                 array(\PDO::PARAM_INT, \PDO::PARAM_INT)
             ),
+            // Test parameters that require a different binding type provided by Type::getBindingType()
+            array(
+                "SELECT * FROM Foo WHERE foo IN (?)",
+                array(array('oof', 'rab')),
+                array('customReverseType'),
```

Comment:
CustomReverseType isn't a binding type, but a custom type that extends `Doctrine\DBAL\Types\Type` and is added using `Type::addType('customReverseType', 'Namespace\To\Types\CustomReverseType');`.  
It's PDO binding type is retrieved by calling `getBindingType()` this happens in `Doctrine\DBAL\Connection->getBindingInfo().  

This is especially important if you both rely on a binding type of PARAM_LOB and have implemented `convertToDatabaseValue`, I'm using it like this:
```
...
 ->setParameter("id", array(binary1, binary2, binaryN), CustomBinaryType::CUSTOMBINARYTYPE);
```